tests/integration/test_fill.py

```diff
@@ -19,21 +19,69 @@
 import cunumeric as num
 
 
-def test_basic():
-    anp = np.random.randn(4, 5)
-    bnp = np.full((4, 5), 13)
-    a = num.array(anp)
-    b = num.array(bnp)
+def test_fill_empty_array():
+    a_np = np.array([])
+    a_num = num.array(a_np)
+    a_np.fill(1)
+    a_num.fill(1)
+    assert np.array_equal(a_np, a_num)
 
-    anp.fill(13)
-    a.fill(13)
 
-    assert np.array_equal(a, anp)
+def test_fill_none():
+    a_np = np.random.randn(2, 3)
+    a_num = num.array(a_np)
+    a_np.fill(None)
+    a_num.fill(None)
+    assert np.array_equal(a_np, a_num, equal_nan=True)
 
-    bnp.fill(13)
-    b.fill(13)
 
-    assert np.array_equal(b, bnp)
+def test_fill_float_to_int():
+    a_np = np.full((2, 3), 1)
+    a_num = num.array(a_np)
+    a_np.fill(0.6)
+    a_num.fill(0.6)
+    assert np.array_equal(a_np, a_num)
+
+
+def test_fill_negative_float_to_int():
+    a_np = np.full((2, 3), 1)
+    a_num = num.array(a_np)
+    a_np.fill(-1.3)
+    a_num.fill(-1.3)
+    assert np.array_equal(a_np, a_num)
```

Comment:
These could potentially be combined using `parametrize`. I think it might be good to test a wider range of cases (floating zero, floating types wider/narrower than the original int type, etc.) 
```
FLOAT_FILL_VALUES = (-2.4e120, -1.3, 8.9e-130, 0.0, 5.7e-150, 0.6, 3.7e160) # maybe +/- inf and nan too?

@pytest.mark.parametrize("value", FLOAT_FILL_VALUES)
test_fill_float_to_int(value: float) -> None:
    ...
```